core-bundle/src/Resources/public/clipboard.js

```diff
@@ -0,0 +1,35 @@
+/*!
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+window.addEventListener('DOMContentLoaded', function () {
+    document.querySelectorAll('[data-to-clipboard]').forEach(function (link) {
+        var data = JSON.parse(link.getAttribute('data-to-clipboard'));
+
+        var clipboardFallback = function () {
+            var input = document.createElement('input');
+            input.value = data.content;
+            document.body.appendChild(input);
+            input.select();
+            input.setSelectionRange(0, 99999);
+            document.execCommand('copy');
+            document.body.removeChild(input);
+        }
+
+        link.addEventListener('click', function (event) {
+            event.preventDefault();
+
+            if (navigator.clipboard && navigator.clipboard.writeText) {
+                navigator.clipboard.writeText(data.content).catch(clipboardFallback);
+            } else {
+                clipboardFallback();
+            }
+
+            Backend.openModalWindow(380, data.title, '<div style="padding: 20px 12px">'+data.message+'</div>');
```

Comment:
```suggestion
                navigator.clipboard.writeText(data.content).catch(clipboardFallback).finally(function () {
                    Backend.openModalWindow(380, data.title, '<div style="padding: 20px 12px">'+data.message+'</div>');
                });
            } else {
                clipboardFallback();
                Backend.openModalWindow(380, data.title, '<div style="padding: 20px 12px">'+data.message+'</div>');
            }
```
The modal should not be opened before the promise from `navigator.clipboard.writeText` was resolved.

Reply:
I agree that's technically right, but do you really think we need that overhead for the split-second where the data is asynchronously added to the clipboard? I also inflicts on the opposite direction, the user has no screen action when clicking the button _until_ the data is added to the clipboard. I think that's more annoying than the likeliness of trying to paste before async is finished?

Reply:
Don’t know all browser implementations, but some may be asking the user if they allow access to the clipboard. Showing a success message before the user clicked on “allow” is confusing IMO.

I’m fine either way. I just noticed this when reviewing the code...